### PR TITLE
Skip cancel test plan for "NIGHTLY" test plan type

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
This change is to address the "az login" timeout after 24 hours issue.

#### How did you do it?
The previous fix to run az login again using System.AccessToken is not working. The reason is that the System.AccessToken has expired when "az login" is executed in test_plan.py.

This change takes advantage of the "timeout" argument of test_plan.py. If test_plan.py has been running for like 20 hours, exit early with a different return code.

In the pipeline yaml, check return code of test_plan.py. If it is 0 or 2 (polling timeout), consider it as success.

The cancel test plan step still run with condition "always()". This step is updated to check TEST_PLAN_TYPE, if it is nightly, exit early to skip cancel test plan because the test plan could still be running.

#### How did you verify/test it?.
Use draft PR to verify the change.

### CODE REVIEW

File: .azure-pipelines/run-test-elastictest-template.yml [hunk 1]
```diff
@@ -395,14 +377,17 @@ steps:
       inlineScript: |
         set -e
         echo "Try to cancel test plan $TEST_PLAN_ID, cancelling finished test plan has no effect."
+
+        # If TEST_PLAN_TYPE is NIGHTLY, skip the cancel step
```
Comment (yejianquan) [inline .azure-pipelines/run-test-elastictest-template.yml:381]: PR test is possible that pass an empty string, will it affect the logic of PR tests?

File: .azure-pipelines/run-test-elastictest-template.yml [hunk 2]
```diff
@@ -338,10 +326,12 @@ steps:
             echo -n "$(FRONTEND_URL)/scheduler/testplan/$TEST_PLAN_ID "
             echo -e "\033[33mfor detailed test plan progress \033[0m"
             # When "EXECUTING" finish, it changes into "KVMDUMP", "FAILED", "CANCELLED" or "FINISHED"
-            echo "[test_plan.py] poll EXECUTING status"
-            python ./.azure-pipelines/test_plan.py poll -i $TEST_PLAN_ID --expected-state EXECUTING --expected-result ${{ parameters.EXPECTED_RESULT }}
+            echo "[test_plan.py] poll EXECUTING status, timeout 22 hours"
```
Comment (yejianquan) [inline .azure-pipelines/run-test-elastictest-template.yml:329]: Looks like the skip logic in the cancel test plan stage is enough, why we need to change this?

**Review by yejianquan — APPROVED** [review]
LGTM

File: .azure-pipelines/run-test-elastictest-template.yml (hunk 1 above)
Comment (wangxin) [reply .azure-pipelines/run-test-elastictest-template.yml:381]: No, empty string is considered as non-NIGHTLY.

**Pull request merged**